### PR TITLE
ISS 52 report enhancement

### LINKED ISSUE #52: Test Request/Response in interactive report
Currently interactive reports display the individual steps pass/failure in specific test. 
It would be nice to have requests/responses at the step level displayed as well.

### PR DESCRIPTION
**Pull request by santhoshTpixler**
Fixes #52 
More info can be found in the issue thread.

### CODE REVIEW

File: core/src/main/java/org/jsmart/zerocode/core/report/ZeroCodeReportGeneratorImpl.java [hunk 1]
```diff
@@ -93,7 +95,7 @@ public void generateExtentReport() {
         treeReports.forEach(thisReport -> {
 
             thisReport.getResults().forEach(thisScenario -> {
-                ExtentTest test = extentReports.createTest(thisScenario.getScenarioName());
+                ExtentTest test = extentReports.createTest(onlyScenarioName(thisScenario.getScenarioName()));
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/report/ZeroCodeReportGeneratorImpl.java]: - Currently it allows to put the `@@author` anywhere e.g. at the start or end of the scenarioName free text. Also in the middle(but generally users put in the beginning or end of the field.

`onlyScenarioName` will strip everything after the `@@author` mark.

Better, let's keep the original one : `thisScenario.getScenarioName()`

**Comment by authorjapps** [comment]
all looks good, ready to merge. Please take of the `author marker` comment(minor one)

File: core/src/test/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImplE2ETest.java [hunk 2]
```diff
@@ -52,7 +53,7 @@ private void testStepSize(File thisFile) {
             assertThat(rawJsonReport.getResults().get(0).getSteps().size(), is(2));
```
Comment (authorjapps) [inline core/src/test/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImplE2ETest.java]: 👍

File: core/src/main/java/org/jsmart/zerocode/core/report/ZeroCodeReportGeneratorImpl.java (hunk 1 above)
Comment (santhoshTpixler) [reply core/src/main/java/org/jsmart/zerocode/core/report/ZeroCodeReportGeneratorImpl.java]: The original code added the author name to the author field of the extent report as well as the scenario name. It was redundant information. Also removing the author name from the scenario name made it look cleaner. Anyway, the author name is present in each test which is useful for filtering. 
Maybe we can modify the onlyScenarioName implementation to not strip away anything related to scenarion name except the author name. 

image

Comment (santhoshTpixler) [reply core/src/main/java/org/jsmart/zerocode/core/report/ZeroCodeReportGeneratorImpl.java]: The current way of displaying author name in scenario name
image

Comment (authorjapps) [reply core/src/main/java/org/jsmart/zerocode/core/report/ZeroCodeReportGeneratorImpl.java]: > The current way of displaying author name in scenario name

That's fine for now it helps for exact match for scenario name.

> Maybe we can modify the onlyScenarioName implementation to not strip away anything related to scenarion name except the author name.

Yes, for now, let's not do any logic inside the method. We can implement something later after discussion if anything makes life easy.

**Pull request merged**